tests/loaders/test_mqtt.py

```diff
@@ -68,21 +69,10 @@ def read(self, *args) -> MockSeekMessage | None:
 
 
 @pytest.fixture
-def mock_paho(monkeypatch: pytest.MonkeyPatch) -> Iterator[MagicMock]:
-    with monkeypatch.context() as m:
-        mock_paho = MagicMock()
-        m.setitem(sys.modules, "paho", mock_paho)
-        m.setitem(sys.modules, "paho.mqtt", mock_paho.mqtt)
-        m.setitem(sys.modules, "paho.mqtt.client", mock_paho.mqtt.client)
+def mock_client(monkeypatch: pytest.MonkeyPatch) -> None:
+    import paho.mqtt.client as mqtt
 
-        yield mock_paho
-
-
-@pytest.fixture
-def mock_client(mock_paho: MagicMock) -> Iterator[MagicMock]:
-    mock_client = MQTTMock()
-    mock_paho.mqtt.client.Client.return_value = mock_client
-    yield mock_client
+    monkeypatch.setattr(mqtt, "Client", MQTTMock)
```

Comment:
Could you move the import to the generic imports and do the `monkeypatch()` using a `with ...:` in `test_remote_loader_stream()`?

Reply:
- Moved the import to module scope
- Moved applying of the mock to the test.
- Apparently, it is unnecessary to use a context manager. From the monkeypatch documentation:

https://docs.pytest.org/en/stable/how-to/monkeypatch.html

"All modifications will be undone after the requesting test function or fixture has finished. The raising parameter determines if a KeyError or AttributeError will be raised if the target of the set/deletion operation does not exist."